openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -434,6 +439,8 @@ def _objfunc(self, dv_dict):
                     model._clear_iprint()
                     fail = 1
 
+                self._post_run_model_debug_print()
```

Comment:
You may not even need to ever call this in the drivers, if you instead call it in the _exit_ function of the Recording context manager.

Reply:
I was wondering about that but Justin told me "Print the des-vars before it calls the model, print the objective, constraints after the model returns"

Reply:
maybe in case the model dies?

Reply:
Yeah, if it outright dies, then I can see a need for recording them first.

Reply:
but I still think the pre function and the post function can be a called at the appropriate points in the context manager functions, so you never have to put them in drivers.

Reply:
good idea about the context manager. I didn't know if we wanted to hide it, especially in something named "Recording". I suppose we could rename it to something else. "Debugging" ? "RecordingDebugging" ? Or am I worrying over nothing?